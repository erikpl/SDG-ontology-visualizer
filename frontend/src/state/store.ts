import { combineReducers, createStore } from 'redux';
import apiErrorReducer from './reducers/apiErrorReducer';
import ontologyReducer from './reducers/ontologyReducer';
import fullscreenReducer from './reducers/fullscreenReducer';
import loginReducer from './reducers/loginReducer';
import languageReducer from './reducers/languageReducer';

const rootReducer = combineReducers({
  apiError: apiErrorReducer,
  ontology: ontologyReducer,
  fullscreenStatus: fullscreenReducer,
  login: loginReducer,
  languages: languageReducer,
});

const store = createStore(rootReducer);

export type RootState = ReturnType<typeof rootReducer>;

export default store;
