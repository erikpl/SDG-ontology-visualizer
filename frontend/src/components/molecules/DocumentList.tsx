import { Box, Flex, Heading, Stack } from '@chakra-ui/react';
import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { getDocumentsForSubgoal } from '../../api/ontologies';
import { RootState } from '../../state/store';
import { Document } from '../../types/ontologyTypes';
import DocumentBox from '../atoms/DocumentBox';

const DocumentList: React.FC = () => {
  const [docList, setDocList] = useState<Array<Document>>([]);
  const selectedSubgoal = useSelector((state: RootState) => state.ontology.selectedSubGoal);
  const langList = ['ENG', 'DAN', 'NLD', 'ITA', 'DEU'];
  const pageNum = 1;
  
  const loadDocuments = async () => {
    if (!selectedSubgoal) return;
    const data = await getDocumentsForSubgoal(selectedSubgoal.SubjectLabel, langList, pageNum);
    setDocList(data);
  };

  useEffect(() => {
    loadDocuments();
  }, [selectedSubgoal]);

  if (!docList || docList.length === 0) {
    return <Box height="200px" />;
  }

  return (
    <Stack align="center" spacing="7">
      <Box align="center" px="10">
        <Heading size="lg" mb="10" color="cyan.700">
          Dokumenter
        </Heading>
      </Box>
      {docList.map((document) => (
        <DocumentBox document={document} />
      ))}
      <Flex justifyContent="space-around">
        Side &nbsp;
        {pageNum}
      </Flex>
    </Stack>
  );
};

export default DocumentList;
