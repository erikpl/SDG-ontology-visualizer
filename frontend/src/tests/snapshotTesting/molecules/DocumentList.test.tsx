import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../../state/store';

import DocumentList from '../../../components/molecules/DocumentList';

it('renders with no items', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <DocumentList />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});