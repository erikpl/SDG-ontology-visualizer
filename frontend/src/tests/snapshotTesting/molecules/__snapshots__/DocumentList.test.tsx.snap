// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with no items 1`] = `
<div
  align="center"
  className="css-tthxno"
>
  <h2
    className="chakra-heading css-1qtlgnn"
  >
    Laster dokumenter...
    <div
      className="chakra-spinner css-1y7joxr"
    >
      <span
        className="css-f8n5zr"
      >
        Loading...
      </span>
    </div>
  </h2>
</div>
`;
